lib/watch-mode-skip-tests.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * The WatchModeSkipTests class is used to determine
+ * if a test should be skipped using filters provided
+ * by the user in watch mode.
+ */
+export class WatchModeSkipTests {
+	/**
+	 * Properties are public to allow
+	 * for easy sending to the worker.
+	 */
+	fileRegexOrNull = null;
+	testRegexOrNull = null;
+
+	constructor(watchModeSkipTestsData = undefined) {
+		if (!watchModeSkipTestsData) {
+			return;
+		}
+
+		this.fileRegexOrNull = watchModeSkipTestsData.fileRegexOrNull;
+		this.testRegexOrNull = watchModeSkipTestsData.testRegexOrNull;
+	}
+
+	shouldSkipFile(file) {
+		if (this.fileRegexOrNull === null) {
+			return false;
+		}
+
+		return !this.fileRegexOrNull.test(file);
```

Comment:
These kinds of invocations can be shortened like so:

```suggestion
		return this.#fileRegex?.test(file) !== true;
```

Reply:
I tested this and it doesn't work.


Counterexample:
```javascript
#fileRegex = null;

this.#fileRegex?.test(file) !== true;
---------------------------
this.#fileRegex?.test(file) // is undefined because #fileRegex is null
undefined !== true // Is true because undefined does not equal to true
// But the correct value is false!
```
This is incorrect because when the the filter is null we should *not* skip anything, the correct result is false.


A shorter version that does work is:
```javascript
!(this.#fileRegex?.test(file) ?? true);
```